crates/humanode-runtime/src/lib.rs

```diff
@@ -1584,6 +1584,8 @@ impl_runtime_apis! {
     #[cfg(feature = "try-runtime")]
     impl frame_try_runtime::TryRuntime<Block> for Runtime {
         fn on_runtime_upgrade(checks: frame_try_runtime::UpgradeCheckSelect) -> (Weight, Weight) {
+            sp_io::init_tracing();
```

Comment:
Why do we need to do manual initialization? I kind of assumed this would be handled by the framework, such that we don't have to; it also seems like this is not the place to do it - of course, unless it is justified by some logic was not communicated anywhere here

Reply:
It's not handled by framework now, tested. As I got, it's just developed to trace just transactions execution by enabling was tracing manually. The same as here - https://github.com/humanode-network/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/client/rpc-api/src/state/mod.rs#L157

Reply:
It's enabled just for some parts of code, not everywhere.

Reply:
Should we correct this at upstream?

Reply:
It's a good question, but it can take a lot of time on agreeing where it should be enabled or not. As I got, their goal  was just tracing transaction logic execution.

Reply:
Looks like we need to do some more research here. It might be that they messed this up - or the while tracing integration is a not fully thought through thing. If they only need it for that tracing transaction logic execution and they don't enable it for `try_runtime` - why should we?

Reply:
Your explanation above - https://github.com/humanode-network/humanode/pull/1325#discussion_r1832709696 sounds right, and I guess we shouldn't enable it. It's enough just use `frame_support::log` for logging `on_runtime_upgrade` execution to log operations if we need it.

Reply:
Seems like adding the feature to the runtime to enable the nested futures is the only takeaway here - cause that's just a strait-up improvement; but let's add it in a separate PR, and close this one as is for historical preservation of the context.